src/app/services/cache/cache.service.ts

```diff
@@ -99,9 +102,43 @@ export class CacheService implements OnDestroy {
 
       if (isCacheChanged) {
         this.orderHistoryCache = _.reverse(_.sortBy(this.orderHistoryCache, 'orderDate'));
-        if (isDefined(this.orderHistorySubject)) {
-          this.orderHistorySubject.next(this.orderHistoryCache);
-        }
+      }
+
+      if (isDefined(this.orderHistorySubject) && this.ordersToMonitor.length !== 0) {
+        this.orderHistorySubject.next(this.orderHistoryCache);
+        setTimeout(() => {
+          this.fetchOrderUpdatesRecursive();
+        }, CacheService.ORDER_POLLING_RATE);
+      }
+    });
+  }
+
+  private fetchIncomingBatchesRecursive(): void {
+    this.messageService.getIncomingBatches().then((batches: Batch[]) => {
+      let isCacheChanged = false;
+      if (batches.length !== this.batchUpdatesCache.length) {
+        isCacheChanged = true;
+      } else {
+        batches.forEach((batch: Batch) => {
+          const i = _.findIndex(this.batchUpdatesCache, (cachedBatch: Batch) => {
+            return cachedBatch.id === batch.id;
+          });
+          if (this.batchUpdatesCache[i].state !== batch.state || this.batchUpdatesCache[i].batchEta !== batch.batchEta) {
```

Comment:
So atm we have no eta shit implemented on the back end. "IFF" we get the going post demo is that gonna break this check? I mean im doubtful that we get eta going but just wanna confirm

Reply:
It shouldn't, ideally the ETA only changes if something goes wrong/when we know that something is delayed. So shouldn't redraw too much